### PR TITLE
Adds instructions for OAuth

### PR DESCRIPTION
**Pull request by jmnsf**
This matches the implementation details of both what's committed already and of the changes from PRs https://github.com/getconversio/conversio-app/pull/3800, https://github.com/getconversio/conversio-app/pull/3799, https://github.com/getconversio/conversio-app/pull/3797 and https://github.com/getconversio/conversio-app/pull/3789. So if those PRs change significantly, this will have to as well. I don't expect that, however, as the spec is generally followed throughout.

I tried to simplify the whole protocol as much as possible here, hopefully it's succinct but detailed enough for understanding.

### CODE REVIEW

File: source/includes/_authentication.md [hunk 1]
```diff
@@ -6,10 +6,239 @@ $ curl "api_endpoint_here" \
...
+
+## API Key Authentication
+
 All paths should prefixed with https://app.conversio.com/api/v1 and will need the following HTTP header:
 
 `X-ApiKey: YOUR_API_KEY`
 
 <aside class="notice">
-Your API Key can be found on the profile page in your Conversio control panel
+Your API Key can be found in your store's <a target="_blank" href="https://app.conversio.com/profile">profile page</a>.
```
Comment (dlebech) [inline source/includes/_authentication.md]: Can't you use `[profile page](link)` here?

File: source/includes/_authentication.md [hunk 2]
```diff
@@ -6,10 +6,239 @@ $ curl "api_endpoint_here" \
...
+* PoP Token: Proof of Possession Token. The only type of Access Token issued by Conversio.
+* Code: A short-lived 60 character string that can be exchanged for an Access Token.
+* Client Secret: A 60 character string that acts as password when Authenticating a Partner App and as symmetric key when signing a PoP Token.
+* JWS: [JSON Web Signature](https://tools.ietf.org/html/rfc7515)
+
+### Issuing Access Token
+
+Conversio uses the Authorization Code Grant flow from OAuth 2.0 to issue Access Tokens. This involves three requests:
+
+1. Redirecting the User's to Conversio, where they can authorize the Partner App
```
Comment (dlebech) [inline source/includes/_authentication.md]: User's => Users

File: source/includes/_authentication.md (hunk 1 above)
Comment (jmnsf) [inline source/includes/_authentication.md]: hah, yeah, I was in HTML mode

**Comment by jmnsf** [comment]
cc @dlebech

**Pull request merged**